annif/corpus/document.py

```diff
@@ -61,15 +64,24 @@ def opener(path):
                 return gzip.open(path, mode='rt')
         else:
             opener = open
-
-        with opener(self.path) as tsvfile:
-            for line in tsvfile:
-                text, uris = line.split('\t', maxsplit=1)
-                subjects = [annif.util.cleanup_uri(uri)
-                            for uri in uris.split()]
-                yield self._create_document(text=text,
-                                            uris=subjects,
-                                            labels=[])
+        try:
+            with opener(self.path) as tsvfile:
+                for line in tsvfile:
+                    yield from self._parse_tsv_line(line)
+        except FileNotFoundError as err:
+            raise AnnifException(str(err))
```

Comment:
I think it would be better to define a new AnnifException subclass for this purpose. AnnifException is meant to be a common superclass of all Annif exceptions, not to be directly instantiated. (In fact it could, and maybe should, be turned into an abstract base class, which would prevent it being used like this)

Reply:
Would it be actually better to let Click handle the nonexistent paths in issue #318 and remove this try block?

Reply:
Yes, that's better!